nav2_docking/opennav_docking/src/controller.cpp

```diff
@@ -39,6 +48,22 @@ Controller::Controller(const rclcpp_lifecycle::LifecycleNode::SharedPtr & node)
     node, "controller.v_angular_max", rclcpp::ParameterValue(0.75));
   nav2_util::declare_parameter_if_not_declared(
     node, "controller.slowdown_radius", rclcpp::ParameterValue(0.25));
+  nav2_util::declare_parameter_if_not_declared(
+    node, "controller.use_collision_detection", rclcpp::ParameterValue(true));
```

Comment:
Q: Should this be default on or off?

Reply:
^ might be why CI is having a test fail

Reply:
Good question. I set default to true because I think it's important not to collide with objects. 😅

Which brings me to the next question regarding testing. any advice on how to test collision_checker_->isCollisionFree()?

Reply:
That should be already tested by the costmap topic collision checker's own tests, so you can assume internally it works for the purpose of these tests. If you want to test that the docking works with collision checking (probably wise), you'd need to publish a local costmap / footprint in the test so that the collision checker object can obtain it. You could do something like dock 1x with higher but valid cost, then dock 1x with some legal points in the way between the staging and dock pose, then dock 1x with lethal at the dock (which would be the normal use case since it would pick up the dock itself)